packages/auth/src/grant/routes.ts

```diff
@@ -412,6 +413,28 @@ async function continueGrant(
     accessToken
   })
 }
+async function deleteGrant(
+  deps: ServiceDependencies,
+  ctx: DeleteContext
+): Promise<void> {
+  const { id: continueId } = ctx.params
+  const continueToken = (ctx.headers['authorization'] as string)?.split(
+    'GNAP '
+  )[1]
+  if (!continueId || !continueToken) {
+    ctx.throw(401, { error: 'invalid_request' })
+  }
+  const grant = await deps.grantService.getByContinue(continueId, continueToken)
+  if (!grant) {
+    ctx.throw(404, { error: 'unknown_request' })
+  }
+
+  const deletion = await deps.grantService.deleteGrant(continueId)
```

Comment:
I'm not crazy about the separate lookup and delete, but it looks like that's also how we currently handle grant choices above.

Reply:
I think this will change when we have the middleware that checks that the token belongs to the grant. 
* #882